### DISCUSSION TITLE
[imagickWithLogo.php] Logo is replacing area in center that causing generated QR not working

### DISCUSSION THREAD

**Discussion by leojinno1** [chillerlan/php-qrcode#273] [Q&A]
I'm using example 'imagickWithLogo.php' to generate QR code but look like the logo replace the dot in center area. This may lead to when I set logoSpaceWidth, logoSpaceHeight is quite big, generated QR not working any more. 
- Version addLogoSpace=true (not working): 
![image](https://github.com/chillerlan/php-qrcode/assets/140029933/98e7e634-c7b8-485f-b556-892ba31b6ac7)
- Version addLogoSpace=false:
![image](https://github.com/chillerlan/php-qrcode/assets/140029933/a2dd2d17-b689-4d21-89f0-b7401a1efd8e)

Is there any way that we can reserve blank area for logo without overlapping QR area or effect with QR functioning?

**Comment by codemasher** [comment]
> Is there any way that we can reserve blank area for logo without overlapping QR area or effect with QR functioning?

Hi, the issue with the logo space is that it is taking away from the reserved error correction margin. The logo space size is internally limited to ~25% of the total modules in the matrix (30% is the maximum margin for ECC level H), however, it is not guaranteed that the QR Code remains readable in all cases.

A simple "fix" is to increase the QR Code version and decreasing the logo size to increase the error correction margin (you will need to add the code to proportionally scale the logo space, [the SVG example](https://github.com/chillerlan/php-qrcode/blob/0ca3d2377f8d01b788c738d46f9bf72d9b17083d/examples/svgWithLogo.php#L33-L36) does that). You can specify a minimum version `QROptions::$versionMin = 7` for the auto detection, to ensure the QR Code remains readable.
